src/main/java/org/b612foundation/adam/opm/OdmFormatter.java

```diff
@@ -479,13 +509,85 @@ private static CovarianceMatrix parseLongFormCovariance(List<String> lines)
     return result;
   }
 
+  /**
+   * Parses the long form of the covariance matrix (each element on a separate line with a variable
+   * name). Removes parsed lines from the list. The long form is used in OPM and OMM. OEM uses a
+   * different (short) format for the same data.
+   */
+  private static KeplerianCovariance parseLongFormKeplerianCovariance(List<String> lines)
+      throws OdmParseException {
+    KeplerianCovariance result = new KeplerianCovariance();
+
+    // Covariance reference frame may be omitted, in which case the reference
+    // frame from the
+    // metadata section is assumed.
+    if (containsNext(lines, COV_REF_FRAME)) {
+      String frame = extractField(lines, COV_REF_FRAME);
+      result.setCovRefFrame(parseReferenceFrame(frame));
+    }
+    result.setCAA(extractDoubleNoUnits(lines, CA_A));
+
+    result.setCEA(extractDoubleNoUnits(lines, CE_A));
+    result.setCEE(extractDoubleNoUnits(lines, CE_E));
+
+    result.setCIA(extractDoubleNoUnits(lines, CI_A));
+    result.setCIE(extractDoubleNoUnits(lines, CI_E));
+    result.setCII(extractDoubleNoUnits(lines, CI_I));
+
+    result.setCOA(extractDoubleNoUnits(lines, CO_A));
+    result.setCOE(extractDoubleNoUnits(lines, CO_E));
+    result.setCOI(extractDoubleNoUnits(lines, CO_I));
+    result.setCOO(extractDoubleNoUnits(lines, CO_O));
+
+    result.setCWA(extractDoubleNoUnits(lines, CW_A));
+    result.setCWE(extractDoubleNoUnits(lines, CW_E));
+    result.setCWI(extractDoubleNoUnits(lines, CW_I));
+    result.setCWO(extractDoubleNoUnits(lines, CW_O));
+    result.setCWW(extractDoubleNoUnits(lines, CW_W));
+
+    boolean hasMeanAnomalyBased = containsLater(lines, CM_M);
+    boolean hasTrueAnomalyBased = containsLater(lines, CT_T);
+    if (hasMeanAnomalyBased && hasTrueAnomalyBased) {
+      throw new OdmParseException("Keplerian Covariance specified both True and Mean anomaly based rows, " +
+          "need to be only one");
+    }
+
+    if (!hasMeanAnomalyBased && !hasTrueAnomalyBased) {
+      throw new OdmParseException("Keplerian Covariance has neither true nor mean anomaly based rows, " +
+          "need at least one");
+    }
+
+    if (hasMeanAnomalyBased) {
+      result.setCMA(extractDoubleNoUnits(lines, CM_A));
+      result.setCME(extractDoubleNoUnits(lines, CM_E));
+      result.setCMI(extractDoubleNoUnits(lines, CM_I));
+      result.setCMO(extractDoubleNoUnits(lines, CM_O));
+      result.setCMW(extractDoubleNoUnits(lines, CM_W));
+      result.setCMM(extractDoubleNoUnits(lines, CM_M));
+    }
+
+    if (hasTrueAnomalyBased) {
+      result.setCTA(extractDoubleNoUnits(lines, CT_A));
+      result.setCTE(extractDoubleNoUnits(lines, CT_E));
+      result.setCTI(extractDoubleNoUnits(lines, CT_I));
+      result.setCTO(extractDoubleNoUnits(lines, CT_O));
+      result.setCTW(extractDoubleNoUnits(lines, CT_W));
+      result.setCTT(extractDoubleNoUnits(lines, CT_T));
+    }
+
+
+
```

Comment:
nit: could you run google-java-format (there's a jar you can [run directly](https://github.com/google/google-java-format#from-the-command-line), or there's an IDEA plugin)?

Reply:
I don't know why this keeps messing up. Let's catch up on Slack to try to fix this once and for all.